Lib/test/test_print.py

```diff
@@ -164,6 +164,32 @@ def test_string_with_stream_redirection(self):
         self.assertIn('Did you mean "print(<message>, '
                 'file=<output_stream>)', str(context.exception))
 
+    def test_print_with_stream_redirection_using_max(self):
```

Comment:
Prompted by that, I'd also suggest another test case: `test_stream_redirection_hint_is_specific_to_rshift` that checks:

    print << sys.stderr

*doesn't* produce the new hint in its error message.